elide-core/src/main/java/com/yahoo/elide/security/PermissionManager.java

```diff
@@ -0,0 +1,464 @@
+                    runPermissionChecks(fieldChecks.get(i), mode, resource, changeSpec);
+                    if (mode == CheckMode.ANY) {
+                        return;
+                    }
+                    hasPassingCheck = true;
+                } catch (ForbiddenAccessException e) {
+                    // Ignore and keep looking or queueing
+                }
+            }
+        }
+
+        // If nothing succeeded, we know nothing is queued up. We should fail out.
+        if (!hasPassingCheck && !hasDeferredChecks) {
+            throw new ForbiddenAccessException();
+        }
+    }
+
+    /**
+     * Execute field-aware checks for a single field by specifying the set of class and field checks to be run.
+     *
+     * @param classChecks Class-level checks
+     * @param fieldChecks Field-level checks
+     * @param classCheckMode Class-level check mode
+     * @param fieldCheckMode Field-level check mode
+     * @param resource Resource to check
+     * @param changeSpec Change spec
+     * @param hasDeferredChecks Whether or not this check can expect deferred checks later on
+     */
+    private void fieldAwareExecute(Class<? extends Check>[] classChecks,
+                                   Class<? extends Check>[] fieldChecks,
+                                   CheckMode classCheckMode,
+                                   CheckMode fieldCheckMode,
+                                   PersistentResource resource,
+                                   ChangeSpec changeSpec,
+                                   boolean hasDeferredChecks) {
+        // Check full object, then field
+        boolean entityFailed = false;
+        if (classChecks != null) {
+            try {
+                runPermissionChecks(classChecks, classCheckMode, resource, changeSpec);
+            } catch (ForbiddenAccessException e) {
+                // Ignore this and continue on to checking our fields
+                entityFailed = true;
+            }
+        }
+
+        if (fieldChecks != null && fieldChecks.length > 0) {
+            try {
+                runPermissionChecks(fieldChecks, fieldCheckMode, resource, changeSpec);
+            } catch (ForbiddenAccessException e) {
+                if (fieldCheckMode == CheckMode.ALL || !hasDeferredChecks) {
+                    // No need to wait if we either (a) require all checks to pass or (b) don't have deferred checks
+                    // to wait on
+                    throw e;
+                }
+            }
+        } else if (entityFailed) {
+            throw new ForbiddenAccessException();
+        }
+    }
+
+    /**
+     * Extracted checks.
+     */
+    @AllArgsConstructor
+    public static final class ExtractedChecks {
+        private final Class<? extends Check>[] anyChecks;
+        private final Class<? extends Check>[] allChecks;
+
+        @SuppressWarnings("unchecked")
+        public Class<? extends Check>[] getAllChecks() {
```

Comment:
The name here is misleading.  Does it mean to return 100% of checks regardless of type OR does it mean to just return the 'all' checks?